apps/aecore/lib/aecore/contract/tx/contract_call_tx.ex

```diff
@@ -254,6 +254,11 @@ defmodule Aecore.Contract.Tx.ContractCallTx do
     DataTx.standard_deduct_fee(accounts, block_height, data_tx, fee)
   end
 
+  @spec gas :: non_neg_integer()
+  def gas do
+    Constants.call_gas_price_multiplier() * GovernanceConstants.block_gas_limit()
```

Comment:
What does this function return? Is it the gas price of transaction? The calculation seems weird.

Reply:
As it was stated in the issue's, PR's, function's  name:

> [**Add get base tx gas function** to all transactions](https://github.com/aeternity/elixir-node/issues/737)
>  [Added **functionality to get a gas price** for all transactions](https://github.com/aeternity/elixir-node/pull/788)
> [def **gas** do](https://github.com/aeternity/elixir-node/pull/788/files/7fc1ee12191921a20062288705abf82fe48e4cea#diff-4d3873e694a8fe0632d13c6224d6aea5R258) ...

It surely **returns**: 
> the gas price of transaction

Regarding calculation: Personally, I don't see any problems with it, if you don't like `*` (multiplier) operation, you are free to replace it with the implementation, for instance, involving `+` operation.

Reply:
I find it weird that gas price of transaction is bigger then block_gas_limit. For example:

for ContractCallTx it's: `Constants.call_gas_price_multiplier() * GovernanceConstants.block_gas_limit()`
where `call_gas_price_multiplier` is 30 so this results with value bigger then block_gas_limit.